shared/src/types/eth_bridge_pool.rs

```diff
@@ -49,6 +55,27 @@ pub struct PendingTransfer {
     pub gas_fee: GasFee,
 }
 
+impl keccak::encode::Encode for PendingTransfer {
+    fn tokenize(&self) -> Vec<Token> {
+        let from = Token::String(self.gas_fee.payer.to_string());
+        let fee = Token::Uint(u64::from(self.gas_fee.amount).into());
+        let to = Token::Address(self.transfer.recipient.0.into());
+        let amount = Token::Uint(u64::from(self.transfer.amount).into());
+        let nonce = Token::Uint(self.transfer.nonce.clone().into());
```

Comment:
How come we don't include `self.transfer.asset` as part of the encoding (which eventually feeds into the hash)?

Reply:
The gas fee payer and amount feeds into the hash of `PendingTransfer`, so it looks like the same `TransferFromEthereum` could be added to the pool multiple times but with different gas fees? This is the case with the current `HashSet` implementation as well I think (which uses the default `Hash` derivation).

Reply:
Good question! I should probably add that.
And yes, the same transfer can be submitted multiple times with different gas fees. I'm not sure exactly why someone would do that, but there is no reason to stop them from doing it.

Reply:
It feels like the `PendingTransfer`s that go through the pool should be unique by nonce, since on the Ethereum smart contract side, we shouldn't expect to have multiple transfers redeemed with the same nonce? I need to familiarize myself still with the nonce/replay protection so not sure if this is correct. Maybe we could allow multiple `PendingTransfer`s with the same nonce (in the `TransferFromEthereum` struct), but only the first one to be redeemed on the smart contract works, later ones fail. In which case we may as well prevent transfers with the same nonce going through the pool somehow, but I'm not sure

re specs: https://github.com/anoma/namada/blob/e88eeac2f7eae59466fce12b22f02d50415fd4f6/documentation/specs/src/interoperability/ethereum-bridge/transfers_to_ethereum.md#L126-L133

Reply:
The first one that uses a nonce in an Ethereum smart contract will pass and the rest fail. It is the smart contracts job to enforce this. One possible scenario is that someone under pays gas fees, so submits with higher fees to get it relayed. Then the cheaper one will time out.

Reply:
But in theory, if you don't pay enough gas, you can submit the same transfer with more gas to get it relayed. Then the older one will time out. And this way we don't have to do complicated replacement / remove from pool user logic.